File: app/models/brewery.rb
# == Schema Information
#
# Table name: breweries
#
#  id           :integer          not null, primary key
#  brewery_name :string
#  address      :string
#  longitude    :float
#  latitude     :float
#  image        :string
#  description  :text
#  website      :string
#  created_at   :datetime         not null
#  updated_at   :datetime         not null
#

class Brewery < ActiveRecord::Base
  has_and_belongs_to_many :tours
  geocoded_by :address
  after_validation :geocode, :if => :address_changed?

end

File: app/assets/javascripts/models/Brewery.js
var app = app || {};

app.Brewery = Backbone.Model.extend({
  urlRoot: "/breweries",
  
  defaults:{
    brewery_name: "",
    address: "",
    longitude: 100,
    latitude: 100,
    image: "",
    description: "",
    website: ""
  },

  initialize: function (){
    console.log( "A new brewery is created" );
  }
});

var brewery = new app.Brewery();

brewery.fetch();
